R/ChromBackendMemory.R

```diff
@@ -0,0 +1,212 @@
+#'     empty), a default `list` of empty `data.frame` with the core peaks
+#'     variables will be created. The length of the list should match the number
+#'     of chromatograms in the `chromData` parameter.
+#'
+#'
+#' @author Philippine Louail
+#'
+#' @exportClass ChromBackendMemory
+NULL
+
+#' @noRd
+setClass("ChromBackendMemory",
+         contains = "ChromBackend",
+         slots = c(chromData = "data.frame",
+                   peaksData = "list"),
+         prototype = prototype(chromData = fillCoreChromVariables(data.frame()),
+                               peaksData = list(.EMPTY_PEAKS_DATA),
+                               readonly = FALSE,
+                               version = "0.1"))
+
+#' @rdname ChromBackendMemory
+setMethod("backendInitialize", "ChromBackendMemory",
+          function(object,
+                   chromData = fillCoreChromVariables(data.frame()),
+                   peaksData = list(.EMPTY_PEAKS_DATA)) {
+
+            # Ensure chromData is a data.frame
+            if (!is(chromData, "data.frame"))
+              stop("'chromData' needs to be a 'data.frame' with the general",
+                   " chromatogram variables")
+
+            n_cd <- nrow(chromData)
+            # If chromData is not empty, fill default values if needed
+            if (n_cd) {
+              if (is.null(chromData$dataStorage))
+                chromData$dataStorage <- "<memory>"
+              # Validate and clean up chromData
+              validChromData(chromData)
+              chromData <- chromData[, !vapply(chromData,
+                                                     function(x) all(is.na(x)),
+                                                     logical(1)), drop = FALSE]
+            } else {
+              chromData <- fillCoreChromVariables(data.frame())
+            }
+
+            # Assign chromData to the object@chromData slot
+            object@chromData <- chromData
+
+            # Handle peaksData
+            if (length(peaksData) > 1) {
+              validPeaksData(peaksData)
+            } else {
+              peaksData <- replicate(n_cd, .EMPTY_PEAKS_DATA, simplify = FALSE)
+            }
+            # Assign peaksData to the object@peaksData slot
+            object@peaksData <- peaksData
+
+            object
+          })
+
+#' @rdname hidden_aliases
+setMethod("backendMerge", "ChromBackendMemory", function(object, ...) {
+  object <- unname(c(object, ...))
+  not_empty <- lengths(object) > 0
+  if (any(not_empty))
+    res <- .df_combine(object[not_empty])
+  else res <- object[[1L]]
+  validChromData(res@chromData)
+  validPeaksData(res@peaksData)
+  res
+})
+
+#' @rdname hidden_aliases
+#' @description This method returns the chromatographic data stored in the
+#' backend. If not specified otherwise it will return all defined column in the
+#' chromData slot as well as dding the coreChromVariables missing with NA values.
+#'
+#' @importMethodsFrom S4Vectors sapply
```

Comment:
do we really need to import `sapply` from `S4Vectors`? note that during Bioconductor package review they will also ask us to replace any `sapply()` with either `lapply()` or `vapply()` - the use of `sapply()` in code is discouraged (because the type of result the function returns can differ depending on the input).